spec/services/experiment_control_service_spec.rb

```diff
@@ -316,5 +315,51 @@
         ExperimentControlService.schedule_daily_stale_patient_notifications(experiment.name, patients_per_day: 1)
       }.to change { Experimentation::TreatmentGroupMembership.count }.by(1)
     end
+
+    it "does not create notifications or update experiment status if today is before the experiment before date" do
+      experiment = create(:experiment, :with_treatment_group, experiment_type: "stale_patients", start_date: 1.day.from_now, end_date: 1.week.from_now)
+      patient1 = create(:patient, age: 80)
+      create(:blood_pressure, patient: patient1, device_created_at: 100.days.ago)
+
+      expect {
+        ExperimentControlService.schedule_daily_stale_patient_notifications(experiment.name)
+      }.not_to change { Experimentation::TreatmentGroupMembership.count }
+      expect(experiment.reload.state).to eq("new")
+    end
+
+    it "changes the experiment state to 'complete' and does not create notifications if today is after the experiment end date" do
+      experiment = create(:experiment, :with_treatment_group, experiment_type: "stale_patients", start_date: 1.week.ago, end_date: 1.day.ago)
+      patient1 = create(:patient, age: 80)
+      create(:blood_pressure, patient: patient1, device_created_at: 100.days.ago)
+
+      expect {
+        ExperimentControlService.schedule_daily_stale_patient_notifications(experiment.name)
+      }.to change { experiment.reload.state }.from("new").to("complete")
+      expect(Experimentation::TreatmentGroupMembership.count).to eq(0)
+    end
+  end
+
+  describe "self.abort_experiment" do
```

Comment:
I always use `#` for instance methods but hadn't noticed that `.` is the convention for class methods. I'll make that change later because it's in multiple places. Fewer keystrokes to do nothing right now. Thanks for the tip, though.